src/Formatter/PolicyFormatter.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mazedlx\FeaturePolicy\Formatter;
+
+use Stringable;
+use Illuminate\Support\Collection;
+use Mazedlx\FeaturePolicy\FeatureGroups\DirectiveContract;
+
+final class PolicyFormatter implements FormatContract, Stringable
+{
+    private readonly Collection $directives;
+
+    public function __construct(array $directives)
+    {
+        $this->directives = collect($directives);
+    }
+
+    public function __toString(): string
+    {
+        return $this->directives
+            ->map(function (DirectiveContract $directive) {
+                $formattedRules = implode(' ', $directive->rules());
+
+                if (count($directive->rules()) === 1) {
+                    return "{$directive->name()}={$formattedRules}";
+                }
+
+                return "{$directive->name()}=({$formattedRules})";
+            })
+            ->when(
+                config('feature-policy.reporting.enabled'),
+                fn (Collection $headers) => $headers->push('report-to=violation-reports')
+            )
```

Comment:
Removing as it should be part of another PR
```suggestion
            }
```